crates/build-helper/src/lib.rs

```diff
@@ -52,3 +52,34 @@ pub fn rebuild_if_embedded_path_changed<P: AsRef<Path>>(path: P) {
         rebuild_if_path_changed(path);
     }
 }
+
+// Target OS for compiling our crates, as opposed to the build script.
+pub fn target_os() -> String {
+    env_var("CARGO_CFG_TARGET_OS").unwrap()
+}
+
+pub fn detect_apple() -> bool {
+    matches!(target_os().as_str(), "ios" | "macos")
```

Comment:
I was wondering if build-helper has some functionality
that must be run at compile time (which is build.rs's runtime).
Doesn't look like it. Else we would want to add something like an assertion,
or a separate crate for things that must not be used at app runtime.